APE6.rst

```diff
@@ -0,0 +1,499 @@
+ECSV writers to emit such comment data.  Relevant comment strings
+should be serialized within the ``meta`` keyword structure.
+
+Beyond the minimal standard, applications are free to
+create a custom data structure as needed using the top-level ``meta``
+keyword.  The specification of a corresponding ``schema`` keyword to
+allow interpretation and validation of the custom data is highly
+encouraged.
+
+Standard keywords are:
+
+``datatype``: list, required
+   Top-level data type specification as a list of column specifiers.
+
+``delimiter``: one-character string, optional, default=``space``
+   Delimiter character used to separate the data fields.  Allowed
+   delimiter values are the single characters ``space`` or ``comma``.
+
+``meta``: structure, optional
+   Table meta-data as an arbitrary data structure consisting
+   purely of data types that can be encoded and decoded with the YAML
+   "safe" dumper and loader, respectively.  Typically the top level
+   structure is a mapping (dict) or list.  Keywords, comments,
+   history and so forth should be part of ``meta``.
+
+``schema``: string, optional
+   This provides a reference to a schema that can allow interpretation
+   and validation of the ``meta`` values and column definitions.
+   Further details of this keyword are TBD and expected in version 1.1
+   of the ECSV standard.
+
+Each column specifier is a dictionary structure with the following keys:
+
+``name``: string, required
+   Column name
+
+``datatype``: string, required
+  Column data type.  Allowed types are: ``bool``, ``int8``,
+  ``int16``, ``int32``, ``int64``, ``uint8``, ``uint16``, ``uint32``,
+  ``uint64``, ``float16``, ``float32``, ``float64``, ``float128``,
+  ``complex64``, ``complex128``, ``complex256``, and ``string``.
+  Some implementations may not support all types.
+
+``unit``: string, optional
+   Data unit (unit system could be part of schema?).
+
+``format``: string, optional
+   C-style formatting specification for outputting column values.
+   This does not imply nor require that the values in this table are
+   formatted accordingly.
+
+``description``: string, optional
+   Text description of column.
+
+``meta``: structure, optional
+   Column meta-data as an arbitrary data structure consisting
+   purely of data types that can be encoded and decoded with the YAML
+   "safe" dumper and loader, respectively.
+
+
+Data details
+^^^^^^^^^^^^^
+
+The data section follows immediately after the header.  Lines in the
+data section consisting only of zero or more whitespace characters
+(space and/or tab) are ignored.
+
+The first non-blank line in the data section must contain the column
+names formatted according to the CSV writer being used.  This allows
+most CSV reader applications to successfully read ECSV files and
+automatically infer the correct column names.  ECSV readers should
+validate that the column names in this line match those in the header.
```

Comment:
What should a reader do if they don't match?  I think a warning is sufficient, so long as the number of columns at least match.  Though what a "warning" means depends entirely on the implementation, and could be optionally upgraded to an error in a "strict" mode.